plugins/woocommerce-admin/docs/features/onboarding-tasks.md

```diff
@@ -4,78 +4,127 @@ The onboarding tasks provides a way to help store owners get their sites quickly
-*   `$task->undo_dismiss()` - Undo dismissal of a task
-*   `$task->is_dismissed()` - Check if a task is dismissed
-*   `$task->snooze()` - Snooze a task for later
-*   `$task->undo_snooze()` - Undo snoozing of a task
-*   `$task->is_snoozed()` - Check if a task has been snoozed
-*   `$task->mark_actioned()` - Mark a task as actioned.  Optional to help determine completion
-*   `$task->is_actioned()` - Check if a task has been actioned
-*   `$task->get_json()` - Get the camelcase JSON for use in the client
-    * `id` (int) - Task ID.
-    * `title` (string) - Task title.
-    * `canView` (bool) - If a task should be viewable on a given store.
-    * `content` (string) - Task content.
-    * `additionalInfo` (object) - Additional extensible information about the task.
-    * `actionLabel` (string) - The label used for the action button.
-    * `actionUrl` (string) - The URL used when clicking the task if no task card is required.
-    * `isComplete` (bool) - If the task has been completed or not.
-    * `time` (string) - Length of time to complete the task.
-    * `level` (integer) - A priority for task list sorting.
-    * `isActioned` (bool) - If a task has been actioned.
-    * `isDismissed` (bool) - If a task has been dismissed.
-    * `isDismissable` (bool) - Whether or not a task is dismissable.
-    * `isSnoozed` (bool) - If a task has been snoozed.
-    * `isSnoozeable` (bool) - Whether or not a task can be snoozed.
-    * `snoozedUntil` (int) - Timestamp in milliseconds that the task has been snoozed until.
-
-#### TaskList
-
-**Arguments**
+### Step 2 – Register the task in JavaScript.
+
+Next, you have to add your task to the tasks list in JavaScript.
+
+```jsx
+import { __ } from '@wordpress/i18n';
+import {
+  WooOnboardingTask,
+  WooOnboardingTaskListItem,
+} from '@woocommerce/onboarding';
+
+const Task = ( { onComplete, task, query } ) => {
+  // Implement your task UI/feature here.
+  return (
+    <div>
+    </div>
+  );
+};
+
+registerPlugin( 'add-task-content', {
+  render: () => (
+    <WooOnboardingTask id="my-task">
+      { ( {
+        onComplete,
+        query,
+        task,
+      } ) => <Task onComplete={ onComplete } task={ task } query={ query } /> }
+    </WooOnboardingTask>
+  )
+```
+
+### Example
+
+You can find an full example of how to add a custom task in the [examples directory](../examples/extensions/add-task/) as a WP plugin.
+
+## Models and classes
+
+### TaskLists
+
+The `TaskLists` class acts as a data store for tasks. It allows you to create, initialize, add tasks, retrieve task lists, and perform other operations related to task management.
+
+#### Methods
+
+- `TaskLists::instance()`: Returns the class instance of the `TaskLists` interface.
+- `TaskLists::init()`: Initializes the task lists. This method should be called to set up the necessary configurations and hooks for task management.
+- `TaskLists::is_experiment_treatment($name)`: Checks if an experiment is the treatment or control.
```

Comment:
I'm a bit hesitant to add this since it's only useful internally, perhaps we could add more info
```suggestion
- `TaskLists::is_experiment_treatment($name)`: Checks if an experiment is the treatment or control. This is internally used by Woo.
```